hoot-josm/src/main/cpp/hoot/core/josm/ops/JosmMapCleaner.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2019, 2020 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "JosmMapCleaner.h"
+
```

Comment:
controls the JOSM based map cleaning from within core; This is the map op you want to add to `map.cleaner.transforms` to get josm cleaning into the convert or conflate pipelines. I'm purposefully not adding it to run by default when josm is enabled due to lack of testing of it so far and the potential increased runtime involved. Possibly, at some point if required we could add a list of josm validators to run by default when josm is enabled, though.